core/tx_pool.go

```diff
@@ -1590,11 +1592,49 @@ func (pool *TxPool) queueTxEvent(tx *types.Transaction) {
 
 // SendTxToSharingClients sends the tx into the pool sharing tx ch and
 // if its full logs it
-func (pool *TxPool) SendTxToSharingClients(tx *types.Transaction) {
-	select {
-	case pool.poolSharingTxCh <- tx:
-	default:
-		pool.logger.Warn("pool sharing tx ch is full")
+func (pool *TxPool) SendTxToSharingClients(tx *types.Transaction) error {
+	pool.sharingClientMu.Lock()
+	defer pool.sharingClientMu.Unlock()
+
+	// If there are no tx pool sharing clients just submit to the local pool
+	if len(pool.config.SharingClientsEndpoints) == 0 || !pool.config.SyncTxWithReturn {
+		err := pool.AddLocal(tx)
+		if err != nil {
+			return err
+		}
+		select {
+		case pool.poolSharingTxCh <- tx:
```

Comment:
Confused here, you have no sharing clients but you're sending to the sharing clients channel?

Reply:
its a noop, this is how we had it previously, i just added a new flag for rpc nodes

Reply:
it will just go through the channel and not do anything if we have a client, i can guard it but its very minimal improvement